mono/metadata/profiler.c

```diff
@@ -743,6 +789,38 @@ mono_profiler_class_loaded (MonoClass *klass, int result)
 	}
 }
 
+void
+mono_profiler_static_field (MonoClassField *field)
+{
+	for (ProfilerDesc *prof = prof_list; prof; prof = prof->next)
+		if ((prof->events & MONO_PROFILE_FIELD_EVENTS) && prof->static_field)
+			prof->static_field (prof->profiler, field);
+}
+
+void
+mono_profiler_static_fields_allocated (MonoClass *klass)
```

Comment:
I've so far been retrieving the pointer with `mono_vtable_get_static_field_data (mono_class_vtable (mono_domain_get (), klass))`, but come to think of it, I'm not actually sure if that's correct in all cases.